src/autoscaler/db/sqldb/storedprocedure_sqldb.go

```diff
@@ -109,17 +109,31 @@ func (sdb *StoredProcedureSQLDb) DeleteAllInstanceCredentials(ctx context.Contex
 	return nil
 }
 
-func (sdb *StoredProcedureSQLDb) ValidateCredentials(ctx context.Context, creds models.Credential) (*models.CredentialsOptions, error) {
+func (sdb *StoredProcedureSQLDb) ValidateCredentials(ctx context.Context, creds models.Credential, appId string) (*models.CredentialsOptions, error) {
 	credOptions := &models.CredentialsOptions{}
 	procedureIdentifier := pgx.Identifier{sdb.config.SchemaName, sdb.config.ValidateBindingCredentialProcedureName}
-	query := fmt.Sprintf("SELECT * from %s($1,$2)", procedureIdentifier.Sanitize())
-	err := sdb.sqldb.QueryRow(ctx, query, creds.Username, creds.Password).Scan(&credOptions.InstanceId, &credOptions.BindingId)
-	if err == sql.ErrNoRows {
-		return nil, nil
-	}
-	if err != nil {
-		sdb.logger.Error("validate-stored-procedure-credentials", err, lager.Data{"query": query, "creds": creds})
+	// 🚸 Due to a programming-error – see definition of function `Create` in
+	// “cred_helper/storedprocedure_cred_helper.go” – we store in each column just the corresponding
+	// app_id. To “mark that”, we use here the `as app_id`-renaming.
+	query := fmt.Sprintf(
+		"SELECT instance_id as app_id, binding_id from %s($1,$2) WHERE binding_id = $3",
```

Comment:
Shouldn't we use the renamed column as well, to make it clear?
```suggestion
		"SELECT instance_id as app_id, binding_id from %s($1,$2) WHERE app_id = $3",
```